### PR TITLE
cp-129: The loader shouldn't be shown instead of form when the user presses sign in/sign up button

### PR DESCRIPTION
**Pull request by HIMDanny**
- Split loading statutes for the authenticated user and sign-up sign-in
- Page loader is not shown when the user presses the submit button
- Button accepts `isLoading` state to show loader conditionally
- Get loading state in sign-in/sign-up forms

## Screenshots

Default state:

image

Loading state:

image

### CODE REVIEW

File: frontend/src/libs/components/button/button.tsx [hunk 1]
```diff
@@ -3,10 +3,16 @@ import styles from './styles.module.scss';
...
   type?: 'button' | 'submit';
+  isLoading?: boolean;
 };
 
-const Button: React.FC<Properties> = ({ type = 'button', label }) => (
-  <button type={type} className={styles['submit']}>
+const Button: React.FC<Properties> = ({
+  type = 'button',
+  label,
+  isLoading,
```
Comment (liza-veis) [inline frontend/src/libs/components/button/button.tsx]:
```suggestion
  isLoading = false,
```

File: frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx [hunk 2]
```diff
@@ -19,6 +19,10 @@ const SignInForm: React.FC<Properties> = ({ onSubmit }) => {
     validationSchema: userSignInValidationSchema,
   });
 
+  const { isLoading } = useAppSelector(({ auth }) => ({
+    isLoading: auth.authenticatedUserDataStatus === DataStatus.PENDING,
```
Comment (liza-veis) [inline frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx]: we should have split loading statuses for authenticated user and sign-up sign-in, create another status like authDataStatus and set it in sign-up /sign-in actions in store
```suggestion
    isLoading: auth.authDataStatus === DataStatus.PENDING,
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx (hunk 2 above)
Comment (HIMDanny) [reply frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx]: sure, no problem

File: frontend/src/slices/auth/auth.slice.ts [hunk 3]
```diff
@@ -23,26 +25,32 @@ const { reducer, actions, name } = createSlice({
   extraReducers(builder) {
     builder.addCase(signUp.pending, (state) => {
       state.authenticatedUserDataStatus = DataStatus.PENDING;
```
Comment (liza-veis) [inline frontend/src/slices/auth/auth.slice.ts]: remove authenticatedUserDataStatus change for signUp/signIn

File: frontend/src/index.tsx [hunk 4]
```diff
@@ -35,16 +35,16 @@ createRoot(document.querySelector('#root') as HTMLElement).render(
...
             ],
           },
+          {
+            path: AppRoute.SIGN_IN,
+            element: <Auth />,
+          },
+          {
+            path: AppRoute.SIGN_UP,
+            element: <Auth />,
+          },
```
Comment (liza-veis) [inline frontend/src/index.tsx]: what is the purpose of this? if it's related to the issue of authenticated-user unnecessary call that you should return this back

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: frontend/src/slices/auth/auth.slice.ts (hunk 3 above)
Comment (HIMDanny) [reply frontend/src/slices/auth/auth.slice.ts]: oh, I got it)

File: frontend/src/index.tsx (hunk 4 above)
Comment (HIMDanny) [reply frontend/src/index.tsx]: yeah, you are right. It is unnecessary now

**Review by liza-veis — APPROVED** [review]

**Review by OlenaYatseniuk — APPROVED** [review]

**Review by Nanaier — APPROVED** [review]

**Review by fshabanov — APPROVED** [review]

**Pull request merged**